pipelines/batch/src/main/java/com/google/fhir/analytics/FetchSearchPageFn.java

```diff
@@ -265,4 +268,40 @@ protected void processBundle(Bundle bundle, @Nullable Set<String> resourceTypes)
       }
     }
   }
+
+  /**
+   * For every URN reference whose target is found in this bundle, updates the reference to the
+   * relative URL of the found resource. This is needed when original references are not URLs
+   * (relative or absolute) and instead are URNs, e.g., `urn:uuid:...`. During parsing, these
+   * references are resolved properly, i.e., if the referenced resource is in the Bundle, it is
+   * found. However, the references are kept as original URNs. With this function we are trying to
+   * simulate the logic of uploading the Bundle to a FHIR server and then downloading those
+   * resources. The logic implemented here is based on how HAPI parser resolves the same references
+   * within a Bundle; see {@code ca.uhn.fhir.parser.ParserState.stitchBundleCrossReferences}.
+   *
+   * @param bundle the bundle whose references are updated.
+   */
+  protected void updateResolvedRefIds(Bundle bundle) throws ProfileMapperException {
```

Comment:
If you do my other suggestion for refactoring `ReadNDJsonFilesFn` into `ReadJsonFilesFn`, please also move this back to that class.

Reply:
Done